Trove.js

```diff
@@ -0,0 +1,340 @@
+
+   You should have received a copy of the GNU Affero General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+function detectWeb(doc, url) {
+	if (url.indexOf('/result?') != -1 || url.indexOf('/newspaper/page') != -1) {
+		return "multiple";
+	} else if (url.indexOf('/newspaper/article') != -1) {
+		return "newspaperArticle";
+	} else if (url.indexOf('/work/') != -1) {
+		return "document";
+	}
+}
+
+function getSearchResults(doc, url) {
+	var items = {};
+	var results;
+	if (url.match(/\/result\?/)) {
+		results = ZU.xpath(doc, "//div[@id='mainresults']//li/dl/dt/a");
+	} else if (url.match(/\/newspaper\/page/)) {
+		results = ZU.xpath(doc, "//ol[@class='list-unstyled articles']/li/h4/a");
+	}
+	for (var i=0; i<results.length; i++) {
+		var url = results[i].href;
+		var title = ZU.trimInternal(results[i].textContent);
+		items[url] = title;
+	}
+	return items;
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, url), function (items) {
+			if (!items) {
+				return true;
+			}
+			var urls = [];
+			for (var i in items) {
+				urls.push(i);
+			}
+			ZU.processDocuments(urls, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+
+}
+
+function scrape(doc, url) {
+	if (url.match(/\/newspaper\/article\//i)) {
+		scrapeNewspaper(doc, url);
+	} else {
+		scrapeWork(doc, url);
+	}
+}
+
+function scrapeNewspaper(doc, url) {
+	// There's a BibTex citation embedded in the page, seems like the easiest way to get the details.
+
+	var bibtex = ZU.xpathText(doc, "//textarea[@id='bibtex-citation']");
+
+	var translator = Zotero.loadTranslator("import");
+	translator.setTranslator("9cb70025-a888-4a29-a210-93ec52da40d4");
+	translator.setString(bibtex);
+
+	// Clean up the BibTex results and add some extra stuff.
+
+	translator.setHandler("itemDone", function (obj, item) {
+		var articleID = item.url.match(/nla\.news-article\d+/)[0];
+		item.itemType = 'newspaperArticle';
+		item.abstractNote = ZU.xpathText(doc, "//meta[@property='og:description']/@content");
+		delete item.type;
+		delete item.itemID;
+
+		// Attaching a PDF is a 2 stage process.
+		// First you need to tell the service to start generating the PDF.
+		// You get back a hash id to include in the PDF url.
+
+		var renditionURL = "/newspaper/rendition/"
+		var prepURL = renditionURL + articleID + '/level/3/prep'
+
+		ZU.HTTP.doGet(prepURL, function(hashID) {
+
+			// It takes some time to generate the pdfs after the prep call.
+			// You can ping the service to see if the PDF is ready,
+			// but when I tried this I was denied permission to the statusText of the response object 
+			// which is necessary because statusText == 'Locked' if the PDF isn't ready.
+			// So I took out the ping call completely and just put in a 5 second pause which seems to work ok.
+			// Note that the service itself uses a 5 second delay if the ping fails.
+
+			var timeout = Date.now() + 5000;
```

Comment:
Yes, this looks problematic and it could lead to larger waiting times.
E.g. the 20 items of [this search](http://trove.nla.gov.au/book/result?l-australian=y&q=wragge) would lead to 100 sec waiting time, which is more than 1,5 minutes!